### PR TITLE
Add back integ tests to distributions

### LINKED ISSUE #14361: elasticsearch integration tests are not run in gradle
cd distribution
gradle check

does nothing... we need to get the rest tests going ASAP or the risk of breaking ES via refactorings in such a way we cannot recover is very high. 

plugin integ tests are working already, i will take a look at distribution/zip and see if i can figure out how to fix it as a first step. any help is appreciated.

### PR DESCRIPTION
**Pull request by rjernst**
rpm and deb are still skipped, but this configures rest tests to run for
zip and tgz distributions

closes #14361

### CODE REVIEW

**Comment by rmuir** [comment]
I first tried this out from distribution/zip, maybe a missing dependency?

```
:distribution:zip:integTest#clean UP-TO-DATE
:distribution:zip:integTest#extract

FAILURE: Build failed with an exception.

* What went wrong:
Cannot expand ZIP '/Users/rmuir/workspace/elasticsearch/distribution/zip/build/distributions/elasticsearch-3.0.0-SNAPSHOT.zip' as it does not exist.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

I'm trying from the top-level build now.

**Comment by rmuir** [comment]
There are also quite a few things missing (just from looking at directory structure) that integ tests did before:

1. using a random tmpdir not rooted under ES to ensure no bogus CWD dependencies: https://github.com/elastic/elasticsearch/blob/2.x/dev-tools/src/main/resources/ant/integration-tests.xml#L23-L26
2. using a space in ES run directory to ensure no bogus URL/file handling:
https://github.com/elastic/elasticsearch/blob/2.x/dev-tools/src/main/resources/ant/integration-tests.xml#L23-L26
3. using a space for any plugin dependencies to check for the same in pluginmanager:
https://github.com/elastic/elasticsearch/blob/2.x/pom.xml#L122

While these may seem "unimportant", I do think they are pretty common problems and these are easy tricks to prevent issues later.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy [hunk 1]
```diff
@@ -57,22 +58,33 @@ class ClusterFormationTasks {
...
+                'pidfile': pidFile,
+                // TODO: make this work for multi node!
+                'discovery.zen.ping.unicast.hosts': "localhost:${config.transportPort}",
+                'path.repo': "${home}/repo",
+                'path.shared_data': "${home}/../",
+                // Define a node attribute so we can test that it exists
+                'node.testattr': 'test',
+                'repositories.url.allowed_urls': 'http://snapshot.test*'
+            ]
+            configFile.setText(esConfig.collect { key, value -> "${key}: ${value}" }.join('\n'), 'UTF-8')
```
Comment (rmuir) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy]: do we have an equivalent for the environment properties being set? These are fairly important, at least JAVA_HOME:

https://github.com/elastic/elasticsearch/blob/2.x/dev-tools/src/main/resources/ant/integration-tests.xml#L162-L165

**Comment by rmuir** [comment]
Works for me. I left some comments, all can be followups.

**Comment by rjernst** [comment]
Thanks for the notes @rmuir! I created a follow up for the issues you mentioned: #14464

**Comment by rjernst** [comment]
And I fixed the dependency issue. Running the tests within `distribution/zip` and `distribution/tar` now works.

**Comment by rmuir** [comment]
+1

**Comment by rjernst** [comment]
I also added the environment variables back in this PR.

**Pull request merged**